src/main/java/io/nats/client/JetStream.java

```diff
@@ -202,7 +203,12 @@
      * @throws TimeoutException if the NATS server does not return a response
      * @throws InterruptedException if the thread is interrupted
      */    
-    public PublishAck publish(Message message, PublishOptions options) throws IOException, InternalError, TimeoutException, InterruptedException;
+    PublishAck publish(Message message, PublishOptions options) throws IOException, InternalError, TimeoutException, InterruptedException;
+
+    CompletableFuture<PublishAck> publishAsync(String subject, byte[] body);
+    CompletableFuture<PublishAck> publishAsync(String subject, byte[] body, PublishOptions options);
+    CompletableFuture<PublishAck> publishAsync(Message message);
+    CompletableFuture<PublishAck> publishAsync(Message message, PublishOptions options);
```

Comment:
This actually is part of https://github.com/nats-io/nats.java/issues/393

Reply:
Looks good, just needs comments.